assets/src/theme/extractPageVariables.js

```diff
@@ -0,0 +1,67 @@
+const balancedVar = require('./balancedVar');
+
+const isSameDomain = ({ href }) => !href || href.indexOf(window.location.origin) === 0;
+const isNotCoreFile = ({ href }) => !href || !href.includes('wp-includes')
+
+const collectRuleVars = (collected, rule, sheet, media = null, supports = null) => {
+  if (rule.type === 1) {
+    [...rule.style].forEach(propertyRaw => {
+      const property = propertyRaw.trim();
+      let remainingValue = rule.style.getPropertyValue(property).trim();
+      let match;
+      while ( (match = balancedVar( remainingValue )) ) {
+        // Split at the comma to find variable name and fallback value.
+        const varArguments = match.body.split( ',' ).map( str => str.trim() );
+
+        // There may be other commas in the values so this isn't necessarily just 2 pieces.
+        // By spec everything after the first comma (including commas) is a single default value we'll join again.
+        const [variableName, ...defaultValue] = varArguments;
+
+        const usage = {
+          selector: rule.selectorText,
+          property,
+          defaultValue: defaultValue.join(','),
+          media,
+          supports,
+          sheet: sheet.href,
+        }
+        if (!(variableName in collected)) {
+          collected[variableName] = { usages: [] };
+        }
+        collected[variableName].usages.push(usage);
+        // Replace variable name (first occurrence only) from result, to avoid circular loop
+        remainingValue = (match.pre || '') + match.body.replace(variableName, '') + (match.post || '');
+      }
+    })
+    return collected;
+  }
+  if (rule.type === 4) {
+    // No nested media queries for now.
```

Comment:
We don't have media queries inside of media queries atm, so I kept it simple for now. Same for `@supports`